File: Algorithm/programmers/고득점키트/완전탐색/소수찾기.py
def isPrime(num):
    if num == 1 or num == 0:
        return False
    for i in range(2, int(num**0.5)+1):
        if not num % i:
            return False
    return True


def go(now, numbers, chk, n, cnt):
    global answer

    if cnt == n+1:
        return

    if now:
        num = int("".join(now))
        if isPrime(num):
            answer.add(num)

    for i in range(n):
        if not chk[i]:
            chk[i] = True
            go(now + [numbers[i]], numbers, chk, n, cnt + 1)
            chk[i] = False

answer = set()
def solution(numbers):
    numbers = list(numbers)
    chk = [False for _ in range(len(numbers))]
    go([], numbers, chk, len(numbers), 0)
    return len(answer)

numbers = "011"
print(solution(numbers))

File: Algorithm/programmers/고득점키트/완전탐색/소수찾기.js
let answer = 0
const combs = new Set()

const isPrime = (number) => {
  if (number < 2) return false
  if (number === 2) return true
  for (let i = 2; i <= Math.sqrt(number); i++) {
    if (number % i === 0) return false
  }
  return true
}

const go = (now, numbers, chk) => {
  const nowNumber = Number(now)
  if (!combs.has(nowNumber)) {
    combs.add(nowNumber)
    if (isPrime(Number(now))) {
      answer += 1
    }
  }

  numbers.forEach((el, i) => {
    if (!chk[i]) {
      chk[i] = 1
      const newNow = now + el
      go(newNow, numbers, chk)
      chk[i] = 0
    }
  })
}

function solution(numbers) {
  const numbersArray = numbers.split("")
  const chk = [...Array(numbersArray.length)].map(() => 0)
  go("", numbersArray, chk)
  return answer
}

const numbers = "011"
console.log(solution(numbers))